src/python/pants/task/simple_codegen_task.py

```diff
@@ -197,7 +197,7 @@ def _do_validate_sources_present(self, target):
     return True
 
   def _get_synthetic_address(self, target, target_workdir):
-    synthetic_name = target.id
+    synthetic_name = target.name
```

Comment:
I replied to your comment about the error: I was wrong about where to fix this.

> So, in that case, I think that's because the name of the workunit is too long in compile.go, here: https://github.com/pantsbuild/pants/blob/master/contrib/go/src/python/pants/contrib/go/tasks/go_compile.py#L106 ... moving that to something shorter (or summarizing into a package name) might help.

And looking around a bit, it looks like `GoTarget.import_path` might be a good choice for a shorter name to use in `compile.go`?